public/app/features/auth-config/ProviderConfigPage.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { connect, ConnectedProps } from 'react-redux';
+
+import { AppEvents, NavModelItem } from '@grafana/data';
+import { getAppEvents, getBackendSrv, isFetchError } from '@grafana/runtime';
+import { Button, Field, Input, InputControl, LinkButton, SecretInput, Select, Stack, Switch } from '@grafana/ui';
+import { FormPrompt } from 'app/core/components/FormPrompt/FormPrompt';
+import { Page } from 'app/core/components/Page/Page';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+
+import { StoreState } from '../../types';
+
+import { fieldMap, fields } from './fields';
+import { loadSettings } from './state/actions';
+import { SSOProvider, SSOProviderDTO, FieldData } from './types';
+import { dataToDTO, dtoToData } from './utils/data';
+import { isSelectableValue } from './utils/guards';
+
+const appEvents = getAppEvents();
+const getPageNav = (config?: SSOProvider): NavModelItem => {
+  if (!config) {
+    return {
+      text: 'Authentication',
+      subTitle: 'Configure authentication providers',
+      icon: 'shield',
+      id: 'authentication',
+    };
+  }
+
+  return {
+    text: config.settings.name || '',
+    subTitle: `To configure ${config.settings.name} OAuth2 you must register your application with ${config.settings.name}. ${config.settings.name} will generate a Client ID and Client Secret for you to use.`,
```

Comment:
While testing locally I found that `config.settings.name` is empty for some providers, for example Github, Azuread, but it has the correct value for others: Gitlab, Google.

Reply:
This is the data we get from backend, so not much can be done about it on the frontend. You can check what is being sent from the browser's `Network` tab. For example, this is how google's config looks like: 
 
![Screenshot 2023-12-21 at 12 37 42](https://github.com/grafana/grafana/assets/8878045/000ec593-d427-40b4-b2e1-ea115025f4e9)

Reply:
Actually, I just check this on the ephemeral instance and all the names are displayed correctly. Where do you see the missing names?

Reply:
Got it. I thought we should display the value from the `provider` field from the root of each provider data. The value of `settings.name` is probable empty in many cases, but let keep it like this for now.

Reply:
The issue with using the `provider` field is that the name needs to be properly formatted for display purposes (e.g. `azuredad` should be displayed as `Azure AD`). That's why we need a separate field for that.